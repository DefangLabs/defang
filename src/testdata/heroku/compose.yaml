services:
  postgres:
    image: postgres:15-alpine
  redis:
    image: redis:7-alpine
  app:
    build: .
    environment:
      DATABASE_URL: postgres://postgres:postgres@postgres:5432/app
      REDIS_URL: redis://redis:5739
    depends_on:
      - postgres
      - redis
