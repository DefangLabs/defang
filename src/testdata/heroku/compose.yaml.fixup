{
  "app": {
    "build": {
      "context": ".",
      "dockerfile": "*Railpack"
    },
    "command": null,
    "depends_on": {
      "postgres": {
        "condition": "service_started",
        "required": true
      },
      "redis": {
        "condition": "service_started",
        "required": true
      }
    },
    "entrypoint": null,
    "environment": {
      "DATABASE_URL": "postgres://postgres:postgres@mock-postgres:5432/app",
      "REDIS_URL": "redis://mock-redis:5739"
    },
    "networks": {
      "default": null
    }
  },
  "postgres": {
    "command": null,
    "entrypoint": null,
    "image": "postgres:15-alpine",
    "networks": {
      "default": null
    },
    "ports": [
      {
        "mode": "host",
        "target": 5432,
        "protocol": "tcp"
      }
    ]
  },
  "redis": {
    "command": null,
    "entrypoint": null,
    "image": "redis:7-alpine",
    "networks": {
      "default": null
    },
    "ports": [
      {
        "mode": "host",
        "target": 6379,
        "protocol": "tcp"
      }
    ]
  }
}