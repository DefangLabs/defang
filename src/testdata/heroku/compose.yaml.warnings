 ! service "app": environment "DATABASE_URL" may contain sensitive information; consider using 'defang config set DATABASE_URL' to securely store this value
 ! service "app": missing memory reservation; using provider-specific defaults. Specify deploy.resources.reservations.memory to avoid out-of-memory errors
 ! service "postgres": missing memory reservation; using provider-specific defaults. Specify deploy.resources.reservations.memory to avoid out-of-memory errors
 ! service "postgres": stateful service will lose data on restart; use a managed service instead
 ! service "redis": missing memory reservation; using provider-specific defaults. Specify deploy.resources.reservations.memory to avoid out-of-memory errors
 ! service "redis": stateful service will lose data on restart; use a managed service instead
