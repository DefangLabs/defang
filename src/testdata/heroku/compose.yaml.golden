name: heroku
services:
  app:
    build:
      context: .
      dockerfile: Dockerfile
    depends_on:
      postgres:
        condition: service_started
        required: true
      redis:
        condition: service_started
        required: true
    environment:
      DATABASE_URL: postgres://postgres:postgres@postgres:5432/app
      REDIS_URL: redis://redis:5739
    networks:
      default: null
  postgres:
    image: postgres:15-alpine
    networks:
      default: null
  redis:
    image: redis:7-alpine
    networks:
      default: null
networks:
  default:
    name: heroku_default
